Release Notes/511.md

```diff
@@ -0,0 +1,29 @@
+# Swift Syntax 510 Release Notes
```

Comment:
511

Reply:
Just noticed it as well: https://github.com/apple/swift-syntax/pull/2274